src/lib/services/feature-toggle-service.ts

```diff
@@ -260,24 +260,20 @@ class FeatureToggleService {
         }
 
         if (
-            oneOf(
+            contextDefinition &&
+            contextDefinition.legalValues &&
+            contextDefinition.legalValues.length > 0
+        ) {
+            const valuesToValidate = oneOf(
                 [...DATE_OPERATORS, ...SEMVER_OPERATORS, ...NUM_OPERATORS],
                 operator,
             )
-        ) {
-            if (contextDefinition?.legalValues?.length > 0) {
-                validateLegalValues(
-                    contextDefinition.legalValues,
-                    constraint.value,
-                );
-            }
-        } else {
-            if (contextDefinition?.legalValues?.length > 0) {
-                validateLegalValues(
-                    contextDefinition.legalValues,
-                    constraint.values,
-                );
-            }
+                ? constraint.value
+                : constraint.values;
```

Comment:
Maybe here we could do:
`const valuesToValidate = constraint.value || constraint.values;` but I wanted to retain the same logic. I don't know if we have cases where we are receiving both value and values in constraints